backend/schema/core.ts

```diff
@@ -1,14 +1,23 @@
+enum WeightUnit {g, kg, lb, ton}
+enum DimensionUnit {mm, cm, m, in, ft}
```

Comment:
Not sure if this is comprehensive, feel free to add the unit you have seen from specs.

Reply:
These units look good. But actually, I think a more realistic modeling is that there could be multiple measurements for a single attribute. For example, sometimes we see the weight expressed in the PDF in *both* pounds and kilograms, or the length in both mm and inches.

I think we might want a single top-level attribute like weight, but then instead of a simple primitive value, it has a sub-object that contains both a unit and a value. And it should accept an array of those sub-objects.

The backend step you're referring to can just be ensuring that we calculate from whatever the input is to whatever our canonical column(s) are. And it'll also have to deal with issues like if we read both inches and millimeters, but they don't actually agree.

Reply:
Interesting thought on storing multiple set of weight values to preserve the raw data. I can make the corresponding change on the back-end.

On a side note, do you have idea on how to display such information on the front-end? Should I return a list of weight objects or just a single one?

Reply:
I have updated the schema and the corresponding data on stove. Can you check if this matches what you would expect?

I will make the changes for the rest of appliances once you are happy with the change in the schema.

Reply:
Mentioned this a bit in the other comment, but I think the frontend should pass in an optional param (it can default to imperial since so far RA is focusing on the U.S. market) which expresses what to put units in.